modules/caddyhttp/vars.go

```diff
@@ -88,6 +89,74 @@ func (m VarsMatcher) Match(r *http.Request) bool {
 	return true
 }
 
+// MatchVarRE matches the value of the context variables by a given regular expression.
+//
+// Upon a match, it adds placeholders to the request: `{http.regexp.name.capture_group}`
+// where `name` is the regular expression's name, and `capture_group` is either
+// the named or positional capture group from the expression itself. If no name
+// is given, then the placeholder omits the name: `{http.regexp.capture_group}`
+// (potentially leading to collisions).
+type MatchVarRE map[string]*MatchRegexp
+
+// CaddyModule returns the Caddy module information.
+func (MatchVarRE) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "http.matchers.var_regexp",
```

Comment:
For consistency with the `vars` handler

```suggestion
		ID:  "http.matchers.vars_regexp",
```